### PR TITLE
dot/types, lib/babe/types: migrate babe header to dot types

### LINKED ISSUE #723: investigate and clean up babe types
## Current Behavior

We currently have a `types` package within the `babe` package that has less than 50 lines of code and should probably be added to the `babe` package unless there is a more suitable home.

### LINKED ISSUE #724: migrate test utils to respective packages
## Expected Behavior

The utilities within the root level `tests` directory live within their respective packages.

## Current Behavior

We have a `tests` directory with utilities that are specific to the `babe` and `runtime` packages.

### PR DESCRIPTION
**Pull request by ryanchristo**
## Changes
- migrate babe header to dot types
- migrate babe test util to dot test utils

## Tests:
```
go test ./... -short
```

### Issues:
closes #723, closes #724

### CODE REVIEW

File: dot/core/test_helpers.go [hunk 1]
```diff
@@ -23,15 +23,20 @@ import (
...
 	"github.com/stretchr/testify/require"
 )
 
-// testGenesisHeader is a test block header
-var testGenesisHeader = &types.Header{
+// TestAuthorityDataKey is the default authority hash used for dot tests
+var TestAuthorityDataKey, _ = common.HexToBytes("0xe3b47b6c84c0493481f97c5197d2554f")
+
+// TestGenesisHeader is a test block header
+var TestGenesisHeader = &types.Header{
```
Comment (noot) [inline dot/core/test_helpers.go:39]: why are these exported now? do other packages use them?

File: dot/core/test_helpers.go [hunk 2]
```diff
@@ -23,15 +23,20 @@ import (
...
 	"github.com/ChainSafe/gossamer/lib/runtime"
 	"github.com/ChainSafe/gossamer/lib/trie"
+
 	"github.com/stretchr/testify/require"
 )
 
-// testGenesisHeader is a test block header
-var testGenesisHeader = &types.Header{
+// TestAuthorityDataKey is the default authority hash used for dot tests
+var TestAuthorityDataKey, _ = common.HexToBytes("0xe3b47b6c84c0493481f97c5197d2554f")
```
Comment (noot) [inline dot/core/test_helpers.go:36]: this isn't a hash, it's the location in the trie of where authority data is fetched from via the runtime

change to "is the location of authority data in the storage trie"

File: dot/core/test_helpers.go (hunk 1 above)
Comment (ryanchristo) [reply dot/core/test_helpers.go:39]: I think I was exploring the option and forgot to revert to local. We should export and reuse from a test utils package but not in this pull request. I will revert.

File: dot/core/test_helpers.go (hunk 2 above)
Comment (ryanchristo) [reply dot/core/test_helpers.go:36]: kk, will do

**Pull request merged**